### PR TITLE
etcdserver: use correct http status code for auth error

### LINKED ISSUE #3017: auth: not found should return 404 not 400
```
< HTTP/1.1 400 Bad Request
< Content-Type: application/json
< X-Etcd-Cluster-Id: 7e27652122e8b2ae
< Date: Fri, 19 Jun 2015 21:19:23 GMT
< Content-Length: 45
< 
* Connection #0 to host 127.0.0.1 left intact
{"message":"auth: User asdf does not exist."}
```

### PR DESCRIPTION
**Pull request by xiang90**
auth error should not always be HTTP.StatusBadRequest.

Fix #3017

### CODE REVIEW

File: etcdserver/auth/auth.go [hunk 1]
```diff
@@ -313,13 +312,13 @@ func (s *Store) CreateOrUpdateRole(r Role) (role Role, created bool, err error)
 
 func (s *Store) CreateRole(role Role) error {
 	if role.Role == RootRoleName {
-		return authErr("Cannot modify role %s: is root role.", role.Role)
+		return authErr(http.StatusBadRequest, "Cannot modify role %s: is root role.", role.Role)
```
Comment (yichengq) [inline etcdserver/auth/auth.go]: http.StatusForbidden ?

File: etcdserver/auth/auth.go [hunk 2]
```diff
@@ -331,13 +330,13 @@ func (s *Store) CreateRole(role Role) error {
 
 func (s *Store) DeleteRole(name string) error {
 	if name == RootRoleName {
-		return authErr("Cannot modify role %s: is superuser role.", name)
+		return authErr(http.StatusBadRequest, "Cannot modify role %s: is superuser role.", name)
```
Comment (yichengq) [inline etcdserver/auth/auth.go]: http.StatusForbidden ?

**Comment by yichengq** [comment]
lgtm. defer to @barakmich

**Comment by barakmich** [comment]
LGTM

**Pull request merged**